### PR TITLE
Allow for ARM architectures in DHCP Network Boot options

### PR DESCRIPTION
**Pull request by keithamus**
I noticed https://github.com/opnsense/core/issues/4814 was raised, requesting the ability to specify alternate files for ARM architectures. It's a shame it was closed for lack of activity, but I presume that doesn't mean I cannot raise a patch for it.

This PR adds the architectures `00:10` (ARM 32 bit) as the field `filename32arm` and `00:11` (ARM 64 bit) as the field `filename64arm`. If not specified these default to `filename32`/`filename64` respectively.

This _does_ change the behavior slightly, as right now the current logic means these architectures fall through to using `filename`. I think it would be fine to try to keep that behaviour, if people feel that's important.

This is my first contribution here, I've read the CONTRIBUTING guidelines and I believe I've done everything necessary. I'm happy to license this work under 2-Clause BSD :slightly_smiling_face:

### CODE REVIEW

**Comment by keithamus** [comment]
@fichtner @AdSchellevis thoughts on this PR? Would you like anything more done to this to resolve #4814?

File: src/etc/inc/plugins.inc.d/dhcpd.inc [hunk 1]
```diff
@@ -1085,6 +1085,10 @@ EOD;
                 $dhcpdconf .= "    filename \"{$dhcpifconf['filename64']}\";\n";
                 $dhcpdconf .= "  } else if option arch = 00:09 {\n";
                 $dhcpdconf .= "    filename \"{$dhcpifconf['filename64']}\";\n";
+                $dhcpdconf .= "  } else if option arch = 00:0a {\n";
+                $dhcpdconf .= "    filename \"{$dhcpifconf['filename32arm'] || $dhcpifconf['filename32']}\";\n";
```
Comment (AdSchellevis) [inline src/etc/inc/plugins.inc.d/dhcpd.inc:1089]: shouldn't this be a `??` and outside the string block?

**Comment by AdSchellevis** [comment]
@keithamus it has been busy, I have some doubts about the dhcp conf, but functionally I don't mind to add it. Can you check if it actually generates the correct config or change the source?

Just leaving a reference here where the numbers come from for future reference : https://www.iana.org/assignments/dhcpv6-parameters/dhcpv6-parameters.xhtml#processor-architecture

File: src/etc/inc/plugins.inc.d/dhcpd.inc (hunk 1 above)
Comment (keithamus) [reply src/etc/inc/plugins.inc.d/dhcpd.inc]: The last time I worked on PHP code, 5.3 was the latest version, so please forgive my rustyness.

**Comment by keithamus** [comment]
@AdSchellevis thanks for the comments, I've adjusted the code and tested it. It generates a suitable config IMO.

**Comment by AdSchellevis** [comment]
@keithamus looks good, thanks!

**Pull request merged**